### DISCUSSION TITLE
"cannot connect to chrome" on Linux after I deleted "/tmp/snap-private-tmp"

### DISCUSSION THREAD

**Discussion by iamumairayub** [seleniumbase/SeleniumBase#2579] [Q&A] [answered]
I have been using SB for several months, for different clients, across different servers.

```
with SB(
    uc=True, 
    xvfb=True,
    browser='chrome',
    extension_dir=",".join(extensions),
    ) as sb:
    pass
```

This code was working perfectly for me but

Its 2nd time where my server's `/tmp/snap-private-tmp` grew huge, occupying 80% of server disk, I deleted it.

and now I started getting 

`
selenium.common.exceptions.WebDriverException: Message: unknown error: cannot connect to chrome at 127.0.0.1:9222
(The process started from chrome location /usr/bin/google-chrome is no longer running, so ChromeDriver is assuming that Chrome has crashed.)
`

I tried updating/reinstalling google-chrome but no use.

I tried `headless` args but no success

```
with SB(
    uc=True, 
    xvfb=True,
    browser='chrome',
    extension_dir=",".join(extensions),
    headless=True, #tried False too
    headless2=True, #tried False too
    ) as sb:
    pass
```

I also tried this but same error

```
with Display(visible=0, size=(headers_to_use['screen_width'], headers_to_use['screen_height'])):

    driver = Driver(
        uc=True, 
        browser='chrome',
        extension_dir=",".join(extensions),
    )
```

Here are results of few commands that proves Chrome is there but just Python not being able connect to it.

```
google-chrome --version
Google Chrome 122.0.6261.111
```

```
/usr/bin/google-chrome
[2964081:2964081:0308/090925.961549:ERROR:zygote_host_impl_linux.cc(100)] Running as root without --no-sandbox is not supported. See https://crbug.com/638180.
```

```
lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 20.04.6 LTS
Release:        20.04
Codename:       focal
```

```
python --version
Python 3.10.13
```

**Comment by mdmintz** [comment, marked as answer]
Your `/usr/bin/google-chrome` call had a link to https://crbug.com/638180, which had a useful comment: https://issues.chromium.org/issues/40480798#comment106. So the first time it didn't work, it was because you improperly deleted a necessary directory that was being used by Chrome. Then you had issues because you reinstalled Chrome with different permissions from before (you need a non-root user). That Chromium issue comment had a link to https://code.visualstudio.com/remote/advancedcontainers/add-nonroot-user.

Anyway, this is more of a Linux/Chrome configuration issue than a SeleniumBase one (you did mention your code was working before you made those changes).
